src/toxcore/io_tokio.rs

```diff
@@ -36,37 +106,36 @@ pub fn send_to_bounded<T: Send + 'static, Tx, E: Debug>(tx: &Tx, v: T, timeout:
 {
     Box::new(send_to(tx, v)
         .timeout(timeout)
-        .map_err(|e| Error::new(ErrorKind::Other,
+        .map_err(|e| IoError::new(IoErrorKind::Other,
             format!("Failed to send message: {}", e)
         )))
 }
 
 /// Send item to a sink using reference
-pub fn send_all_to<T: Send + 'static, S, Tx, E: Debug>(tx: &Tx, s: S) -> IoFuture<()>
+pub fn send_all_to<T: Send + 'static, S, Tx, E: Debug>(tx: &Tx, s: S) -> impl Future<Item=(), Error=SendToError> + Send
```

Comment:
Actually I think we don't need to wrap errors for `send_to*` methods. They can return only error that is defined in their signature as `SinkError` and also they are supposed for internal usage only. So I'd only change it's return type to `impl Future<Item=(), Error=E> + Send` and wrap it only on higher level. @kpp do you agree?

Reply:
Then we should mark them as `pub(crate)`. Why not Error=SinkError?

Reply:
ok, agree. Later when we get rid of `IoFuture` we can just make the whole module `binary_io` private.

> Why not Error=SinkError?

Because `SinkError` is a trait, E is generic parameter `E: SinkError`.

Reply:
> So I'd only change it's return type to impl Future<Item=(), Error=E> + Send and wrap it only on higher level.

How would you wrap it?

Reply:
We'll need to wrap it to another `ErrorKind` anyway whether it's our custom error or tokio error. It seems for each module (tcp, dht, lan discovery, etc) it will be it's own ErrorKind.

Reply:
Also I'd suggest to try this approach first before we decide. I'm not sure how convenient it will be.